packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -279,6 +280,17 @@ export class RestApi extends cdk.Construct implements IRestApi {
     });
   }
 
+  /**
+   * Adds a new model resource to the REST API, and returns it
+   * @param name A name for the model.
+   * @param contentType The content type for the model.
+   * @param schema The schema to use to transform data to one or more output formats.
+   * @param description A description that identifies this model.
+   */
+  public addModel(name: string, contentType: string, schema: any, description?: string): Model {
```

Comment:
You should wrap `contentType`, `schema`, and `description?` in a "props" type object.

Also, I would advise against using `any` for the `schema`. You will not be able to export the type from `jsonschema` (if you decide to go with it) because it is not a JSII module... But I guess you could duplicate it (it's not ideal, but it's pragmatic).

Reply:
Yeah. I was definitely trying to add that type safety to the schema before, but was running into those JSII errors. I'll try your suggestion around adding jsonschema to the bundledDependencies.

Reply:
I was toying with creating a duplicate of the JSONschema type. The JSII linting rules were not fond of the `$schema` and `$ref` properties, as they were detected as not conforming to camelCasing. But I guess this could maybe be handled by transforming the input schema into a CFN schema object.